lib/dgram.js

```diff
@@ -173,11 +173,13 @@ function bufferSize(self, size, buffer) {
   if (size >>> 0 !== size)
     throw new errors.TypeError('ERR_SOCKET_BAD_BUFFER_SIZE');
 
-  try {
-    return self._handle.bufferSize(size, buffer);
-  } catch (e) {
-    throw new errors.Error('ERR_SOCKET_BUFFER_SIZE', e);
+  const ctx = {};
+  const ret = self._handle.bufferSize(size, buffer, ctx);
+  if (ret === undefined) {
+    throw new errors.Error('ERR_SOCKET_BUFFER_SIZE',
```

Comment:
Is this for avoiding changing the error type? Maybe a comment because this looks a bit odd at first?

Reply:
Yes, I really dislike how this was done, but I want to save it for a different PR to change it.